File: src/brs/web/api/http/handler/GetAskOrder.java
package brs.web.api.http.handler;

import brs.Asset;
import brs.SignumException;
import brs.Order;
import brs.assetexchange.AssetExchange;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.JSONData;
import brs.web.api.http.common.LegacyDocTag;
import brs.web.api.http.common.ParameterParser;
import com.google.gson.JsonElement;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.JSONResponses.UNKNOWN_ORDER;
import static brs.web.api.http.common.Parameters.ORDER_PARAMETER;

public final class GetAskOrder extends ApiServlet.JsonRequestHandler {

  private final AssetExchange assetExchange;

  public GetAskOrder(AssetExchange assetExchange) {
    super(new LegacyDocTag[] {LegacyDocTag.AE}, ORDER_PARAMETER);
    this.assetExchange = assetExchange;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) throws SignumException {
    long orderId = ParameterParser.getOrderId(req);
    Order.Ask askOrder = assetExchange.getAskOrder(orderId);
    if (askOrder == null) {
      return UNKNOWN_ORDER;
    }
    Asset asset = assetExchange.getAsset(askOrder.getAssetId());
    return JSONData.askOrder(askOrder, asset);
  }

}

File: src/main/java/brs/http/GetAskOrder.kt
package brs.http

import brs.BurstException
import brs.assetexchange.AssetExchange
import brs.http.JSONResponses.UNKNOWN_ORDER
import brs.http.common.Parameters.ORDER_PARAMETER
import com.google.gson.JsonElement
import javax.servlet.http.HttpServletRequest

internal class GetAskOrder internal constructor(private val assetExchange: AssetExchange) : APIServlet.JsonRequestHandler(arrayOf(APITag.AE), ORDER_PARAMETER) {
    override suspend fun processRequest(request: HttpServletRequest): JsonElement {
        val orderId = ParameterParser.getOrderId(request)
        val askOrder = assetExchange.getAskOrder(orderId) ?: return UNKNOWN_ORDER
        return JSONData.askOrder(askOrder)
    }
}
